Branch update ug to current implementation and Fix contact name bug (#269)
AY2526S1-CS2103T-F12-1/tp: docs/UserGuide.md

Format: `help`

## Contact related commands
---
### Adding a contact: `add-contact`

Adds a contact to the StudyCircle contact list.

Format: `add-contact n/NAME p/PHONE_NUMBER e/EMAIL [g/GROUP_INDEX]…​`

`NAME` Constraints:
* case-insensitive <span style="color:grey">(The system displays names in their original case but treats names like “tom” and “TOM” as the same.)</span>
* Should not be blank <span style="color:grey">(A string with only spaces is considered blank also)</span>
* Support max 50 characters
* Support alphanumeric characters
* Support spaces <span style="color:grey">(Consecutive space not allow)</span>
* Support s/o or d/o <span style="color:grey">(An optional segment that must be placed between two names when used)</span>
* Leading and trailing spaces are automatically removed.

`PHONE_NUMBER` Constraints:
* Must contain only digits
* Minimum 8 digits
* Maximum 15 digits

`EMAIL` Constraints:
* is case-sensitive
* Must start with 'e'
* Follow by 7 digit

> * Must end with '@u.nus.edu'

Reviewer: I think this constraint is far too restrictive. Even within NUS it doesn't support faculty domains, such as comp.nus.edu, or friendly mail aliases. Overall I feel there is no real value in not allowing emails of any form, since we never utilise the restricted domain for any feature.

<box type="tip" seamless>

**Tip:** A contact can be added to multiple groups at once by specifying multiple `g/` prefixes
</box>

Examples:
* `add-contact n/John Doe p/98765432 e/e1234567@u.nus.edu` adds a contact with name: John Doe, phone: 98765432, email: e1234567@u.nus.edu to the contact list.
* `add-contact n/Betsy Crowe g/1 e/e1232567@u.nus.edu p/12345678 g/2` adds a contact with name: Betsy Crowe, phone: 12345678, email: e1232567@u.nus.edu to the contact list and add Betsy Crowe to group 1, 2.

### Deleting a contact : `delete-contact`
Deletes the specified contact from the StudyCircle contact list.

Format: `delete-contact CONTACT_INDEX`

* Deletes the contact at the specified `CONTACT_INDEX`.
* The `CONTACT_INDEX` refers to the index number shown in the displayed contact list.
* The `CONTACT_INDEX` **must be a positive integer** 1, 2, 3, …​

Examples:

* `delete-contact 1` deletes the 1st contact in the current displayed contact list.

### Editing a contact : `edit-contact`
Edits the details of the specified contact's name, phone, email and groups.

Format: `edit-contact CONTACT_INDEX [n/NAME] [p/PHONE] [e/EMAIL] [g/GROUP INDEX]...`

`CONTACT_INDEX` Constraints:

> * Must be a positive integer

Reviewer: ```suggestion
* Should match an index in the currently displayed contact list
```

* Both `GROUP_INDEX` and `CONTACT_INDEXES` **must be positive integers** 1, 2, 3, …​
* `CONTACT_INDEXES` can be one or multiple contact indexes
* `CONTACT_INDEXES` are taken from the currently displayed contact list

<box type="tip" seamless>

> **Tip:** You can add one or more contacts to a group by specifying each contact with a `c/` prefix, e.g., `c/1` or `c/1 c/2 c/3`.

Reviewer: ```suggestion
**Tip:** You can add more than one contact to a group by specifying each extra contact with a `c/` prefix, e.g., `c/1 c/2 c/3`.
```
